### ISSUE TITLE
Published docs do not align with released version

### ISSUE THREAD

**Issue by JacksonWrath** [rook/rook#15161]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

- Versioned docs are showing changes from development work that is not yet released.

**Expected behavior:**

- When I select a version in the docs, the documentation accurately reflects what is released.

Example: 

Changes from commit https://github.com/rook/rook/commit/9610407d895b2e2300094bf85e2e8ef7db39dcdf just a few days ago are already showing up in the 1.15 docs despite not being released yet: https://rook.io/docs/rook/v1.15/CRDs/Shared-Filesystem/ceph-filesystem-crd/#metadata

**Comment by travisn** [comment]
Yes, this is the behavior today. Every time there is a merge to the release branch, the docs for that branch are published. While this is confusing temporarily for new features until the next patch release, we publish the docs with every merge because there are often doc fixes that don't have a feature in the release. So we err on the side of publishing the fixes instead of waiting for the release.

**Comment by JacksonWrath** [comment]
_(okay with moving to discussion if appropriate)_

Ah, gotcha, yea that's fair.

I have to say though, it's fairly frustrating; I wasted hours troubleshooting why my CRDs weren't being updated until I eventually stumbled upon this behavior. It makes me not trust the docs anymore, so in the future I'm just going to directly look at the release artifacts instead...

Any thoughts on the following remedies?
- moving docs to a separate repo so they could be updated separately from features
- putting a note somewhere prominent that docs may include unreleased features
  - a better step (but more work) could be to be able to mark sections of docs for a future release with a warning flag/box, then the docs auto-update to remove that flag/box when the release happens
- don't put new features in patch releases

**Comment by travisn** [comment]
I've been mulling this over. Perhaps the following is reasonable to prevent such pain of finding documented features that aren't released yet:
- Only published the docs during official release.
- Create a github action that would publish the docs if maintainers have the need to publish the docs even without a release

@subhamkrai Thoughts? The first point would require always publishing the docs in master, but in release branches only publish the docs during a tagged build, but not the releases after a merge to the release branch.

**Comment by subhamkrai** [comment]
> I've been mulling this over. Perhaps the following is reasonable to prevent such pain of finding documented features that aren't released yet:
> 
> * Only published the docs during official release.
> * Create a github action that would publish the docs if maintainers have the need to publish the docs even without a release
> 
> @subhamkrai Thoughts? The first point would require always publishing the docs in master, but in release branches only publish the docs during a tagged build, but not the releases after a merge to the release branch.

@travisn I would like to go with the first approach as it will require one time changes and less maintenance  for maintainers

**Comment by subhamkrai** [comment]
@travisn I think you already fixed this right?

**Comment by travisn** [comment]
There was an issue in the v1.17.2 release with this change, still need to investigate...
